extensions/trace_propagators/src/main/java/io/opentelemetry/extensions/trace/propagation/JaegerPropagator.java

```diff
@@ -187,6 +208,42 @@ public static JaegerPropagator getInstance() {
     return buildSpanContext(traceId, spanId, flags);
   }
 
+  private static <C> Baggage getBaggageFromHeader(C carrier, Getter<C> getter) {
+    Baggage.Builder builder = null;
+
+    for (String key : getter.keys(carrier)) {
+      if (key.startsWith(BAGGAGE_PREFIX)) {
+        if (key.length() <= BAGGAGE_PREFIX.length()) {
```

Comment:
Since we already checked startsWith I think this can be equals check

Reply:
I'd rather stick to check the length, as it only depends on a numerical comparison.

Reply:
Ah I meant an equal check on the length - if it starts with it, then the length is guarateed to be at least that length and the `<` here looks redundant

Reply:
Done.